src/types/DataGenerator.cpp

```diff
@@ -191,44 +192,51 @@ void DataGenerator::modelGenerator(int nb_criteria, int nb_categories,
   }
 }
 
-// void DataGenerator::loadDataset(std::string fileName) {}
-
 std::tuple<float, Criteria, PerformanceTable>
 DataGenerator::loadModel(std::string fileName) {
+  pugi::xml_document doc = DataGenerator::openXmlFile(fileName);
+
+  if (DataGenerator::getXmlFileType(fileName) != "model") {
+    // if we have a model xml
+    throw std::invalid_argument(
+        "Cannot find any lambda and profile performances in xml file, "
+        "most likely have a dataset xml file");
+  } else {
```

Comment:
nit: the else statement is not necessary here as if you get into the if statement the run of the function will stop anyway :)

Reply:
took it out